lib/temporal/data_converter.rb

```diff
@@ -1,40 +1,73 @@
+require 'temporal/api/common/v1/message_pb'
+require 'temporal/errors'
 require 'temporal/payload_converter'
 
 module Temporal
   class DataConverter
-    def initialize(payload_converter: Temporal::PayloadConverter::DEFAULT)
+    class MissingPayload < Temporal::Error; end
+
+    def initialize(payload_converter: Temporal::PayloadConverter::DEFAULT, payload_codecs: [])
       @payload_converter = payload_converter
+      @payload_codecs = payload_codecs
     end
 
     def to_payloads(data)
       return if data.nil? || Array(data).empty?
 
       payloads = Array(data).map { |value| to_payload(value) }
-      Temporal::Api::Common::V1::Payloads.new(payloads: payloads)
+      Temporal::Api::Common::V1::Payloads.new(payloads: encode(payloads))
     end
 
     def to_payload_map(data)
```

Comment:
For just internal use of search attributes (kinda, that may have to change a tad) and memo, not sure there is value in these public `_map` methods for all users. (sorry, I know some of my comments are from another PR, I just didn't pay enough attention)

Reply:
A `DataConverter` is not intended as a public interface and users are not expected to interact with these in any way. Users are instead expected to customize this behaviour by providing a payload converter, codecs and a failure converter only. Hope that makes sense

Reply:
So `DataConverter` is private and not user-usable? If so, no prob.

Reply:
Yep, exactly, it is initialized within the Client/Worker and only used internally. User shouldn't know that it exists :)